presto-main/src/main/java/com/facebook/presto/operator/scalar/sql/ArraySqlFunctions.java

```diff
@@ -91,6 +91,26 @@ public static String arrayHasDuplicatesVarchar()
         return "RETURN cardinality(array_duplicates(input)) > 0";
     }
 
+    @SqlInvokedScalarFunction(value = "array_least_frequent", deterministic = true, calledOnNullInput = true)
+    @Description("Determines the least frequent elemement in the array. If there are multiple elements, the function returns the biggest element")
+    @TypeParameter("T")
+    @SqlParameter(name = "input", type = "array(T)")
+    @SqlType("array<T>")
+    public static String array_least_frequent()
+    {
+        return "RETURN IF(COALESCE(CARDINALITY(REMOVE_NULLS(input)), 0) = 0, NULL, TRANSFORM(SLICE(ARRAY_SORT(MAP_ENTRIES(ARRAY_FREQUENCY(REMOVE_NULLS(input))),(a, b) -> IF(a[2] > b[2], 1, IF(a[2] = b[2], IF(a[1] < b[1], 1, -1), -1))), 1, 1), x -> x[1]))";
```

Comment:
Did you try transposing and sorting?

array_sort(TRANSFORM(MAP_ENTRIES(m),(k,v)->(v,k)))

Then you don't need the lambda for the comparison

Reply:
yeah it will be good to do a sqlbenchmark to see which is best.

Reply:
Ran the banchmark test. Turned out the non lambda version is 0.7 secs faster than compared to the one with lambda. 

Here is the result without lambda
<img width="2082" alt="Screenshot 2023-10-08 at 14 41 20" src="https://github.com/prestodb/presto/assets/73743434/8e5efebe-6be8-4bba-b5ab-69ff6f4247f6">

Here is the result with lambda
<img width="2088" alt="Screenshot 2023-10-08 at 14 30 29" src="https://github.com/prestodb/presto/assets/73743434/c6931bfb-631c-4918-955a-97508059960e">